portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +31,69 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  
+  /**
+   * Takes in a query object and places its content in a json formatted string
+   * and returns it.
+   */
+  private String getJson(Query commentQuery){
+    String returnStr = "{";
+    int count = 0;
+
+    DatastoreService dataStore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery prepCommentQuery = dataStore.prepare(commentQuery);
+    
+    for(Entity commentEntity : prepCommentQuery.asIterable()){
+      returnStr += "\"comment"+ Integer.toString(count)+"\": " + "\"" + 
+        commentEntity.getProperty("content") + "\", ";
+      count++;
+    }
+
+    //Lobbing off extra characters from the end of string if necessary
+    if(returnStr.length() > 1){
+      returnStr = returnStr.substring(0, returnStr.length() - 2);
+    }
+
+    return returnStr + "}";
+  }
 
   @Override
+  /**
+   * Updating database with new comment submission and reloading page
```

Comment:
Function comments should usually be of the grammar form:
"Updates the database with new comment submissions and reloads the page".

Reply:
Here and below.